src/H5Tpkg.h

```diff
@@ -152,17 +152,35 @@ struct H5T_stats_t {
 
 /* Context struct for information used during datatype conversions */
 typedef struct H5T_conv_ctx_t {
-    H5T_conv_cb_t cb_struct;
-    hid_t         dxpl_id;
-    hid_t         src_type_id;
-    hid_t         dst_type_id;
+    union {
```

Comment:
Most of the changes in this PR related to turning this struct's fields into a union so we can distinguish which fields are valid to access depending on which conversion command is being performed.